lib/ddtrace/buffer.rb

```diff
@@ -56,5 +70,65 @@ def close
         @closed = true
       end
     end
+
+    # Aggregate metrics:
+    # They reflect buffer activity since last #pop.
+    # These may not be as accurate or as granular, but they
+    # don't use as much network traffic as live stats.
+
+    def measure_accept(trace)
+      @buffer_spans += trace.length
+      @buffer_accepted += 1
+      @buffer_accepted_lengths += trace.length
+    rescue StandardError => e
+      Datadog::Tracer.log.debug("Failed to measure queue accept. Cause: #{e.message} Source: #{e.backtrace.first}")
+    end
+
+    def measure_drop(old_trace, new_trace)
+      @buffer_dropped += 1
+      @buffer_spans -= old_trace.length
+      @buffer_accepted_lengths -= old_trace.length
+      @buffer_accepted += 1
+      @buffer_spans += new_trace.length
+      @buffer_accepted_lengths += new_trace.length
```

Comment:
```suggestion
      measure_accept(new_trace)
```

? might be easier in case logic in `measure_accept` changes, don't need to keep both up to date

Reply:
Okay I'll change this.